src/server/life/MapleMonster.java

```diff
@@ -219,15 +219,15 @@ public synchronized void damage(MapleCharacter from, int damage) { // may be poi
             takenDamage.get(from.getId()).addAndGet(trueDamage);
         }
 
-        if (hasBossHPBar()) {
+        if (hasBossHPBar() && isAlive()) {
```

Comment:
Is boss death properly propagated to players? I.e. Boss health bar disappearing.

Reply:
Yes, boss hp bar disappears on kill